protocol/linkpreview/linkpreview.go

```diff
@@ -162,25 +204,35 @@ func (u OEmbedUnfurler) unfurl() (common.LinkPreview, error) {
 		"accept-language": headerAcceptLanguage,
 		"user-agent":      headerUserAgent,
 	}
+
 	oembedBytes, err := fetchBody(u.logger, u.httpClient, oembedURL.String(), headers)
 	if err != nil {
 		return preview, err
 	}
 
-	var oembedResponse OEmbedResponse
-	if err != nil {
-		return preview, err
-	}
+	var oembedResponse OEmbedBaseResponse
+
 	err = json.Unmarshal(oembedBytes, &oembedResponse)
 	if err != nil {
 		return preview, err
 	}
 
-	if oembedResponse.Title == "" {
-		return preview, fmt.Errorf("missing required title in oEmbed response")
+	if oembedResponse.Title != "" {
+		preview.Title = oembedResponse.Title
 	}
 
-	preview.Title = oembedResponse.Title
+	switch oembedResponse.Type {
+	case "photo":
+		handlePhotoOembedType(&preview, oembedResponse)
+
+	case "video":
+		handleVideoOembedType(&preview, oembedResponse)
+
+	case "rich":
+		handleRichOembedType(&preview, oembedResponse)
+	default:
```

Comment:
To be consistent with formatting we should have a blank line before `default`.

Reply:
Done